android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java

```diff
@@ -186,34 +165,18 @@ public void onClick(View v) {
           }
           Tab currentTab = mMainActivity.getActivityTab();
           if (currentTab == null) {
-              assert false;
               return;
           }
           try {
               URL url = new URL(currentTab.getUrl());
-              updateBraveShieldsButtonState(currentTab.getProfile(), url.toString());
               mBraveShieldsMenuHandler.show(mBraveShieldsButton, url.toString(),
                   url.getHost(), currentTab.getId(), currentTab.getProfile());
           } catch (Exception e) {
-              disableBraveShieldsButton();
+              // Do nothing. Just return w/o showing shields popup.
```

Comment:
in that case we will show enabled button on URLs like `chrome://flags` and etc

Reply:
I checked that `isShieldsOnForTab()` returns false properly for those urls.
Also I added more check for protocols by (`isInvalidProtocolForShields`).
In desktop, shields are disabled for some protocols like `ftp` or `file` protocols.
I think android brave also should do like desktop.

Reply:
ok